repository/src/main/java/com/epam/esm/repository/impl/TagRepositoryImpl.java

```diff
@@ -0,0 +1,97 @@
+package com.epam.esm.repository.impl;
+
+import com.epam.esm.entity.Tag;
+import com.epam.esm.repository.TagRepository;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+import java.io.Serializable;
+import java.util.List;
+import java.util.Optional;
+
+
+@Repository
+public class TagRepositoryImpl implements TagRepository, Serializable {
+
+
+    @Autowired
+    private SessionFactory manager;
+
+    @Override
+    public List<Tag> getAllEntity(int limit, int offset) {
+        Session session = manager.getCurrentSession();
+        return session.createQuery("select t from Tag t order by t.id", Tag.class)
+                .setMaxResults(limit)
+                .setFirstResult(offset)
+                .getResultList();
+
+    }
+
+    @Override
+    public void addEntity(Tag tag) {
+        Session session = manager.getCurrentSession();
+        session.saveOrUpdate(tag);
+    }
+
+    @Override
+    public Optional<Tag> getEntityById(long id) {
+        Session session = manager.getCurrentSession();
+        return session
+                .createQuery("select t from Tag t where t.id = :id", Tag.class)
+                .setParameter("id", id).uniqueResultOptional();
+    }
+
+    @Override
+    public void deleteEntity(long id) {
+        Session session = manager.getCurrentSession();
+        session.createQuery("delete from Tag where id =:id")
+                .setParameter("id", id)
+                .executeUpdate();
+    }
+
+    @Override
+    public void updateEntity(Tag tag) {
+        Session session = manager.getCurrentSession();
+        session.merge(tag);
+
+    }
+
+    @Override
+    public List<Tag> getOnlyTags() {
+        Session session = manager.getCurrentSession();
+        return session.createQuery("select t from Tag  t order by t.id", Tag.class).getResultList();
```

Comment:
Why do you need order by clause?

Reply:
Made. Did not check himself, in a draft version he may have experimented. It's not really necessary here